### DISCUSSION TITLE
Livewire Broadcast Event passing empty model

### DISCUSSION THREAD

**Discussion by robs007** [livewire/livewire#8976] [1. Help]
I'm using broadcast events and reverb with Laravel Echo. Larvel 11. I'm trying to pass a model in the payload, but I get an empty model when it gets to livewire. Maybe I'm accessing something incorrectly. If I pass a simple string variable, it works fine. If I use a standard Laravel listener, it passes the model fine. Not sure what I'm missing. Thanks

```
class StatusSMS implements ShouldBroadcastNow
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    public string $name;

    public function __construct(public ResStudent $resStudent )
    {
        $this->name = 'Bill Smith';
    }

    public function broadcastOn(): Channel
    {
        return  new Channel('sms-message-status');
    }
}
```
Livewire component

``` 
 public function getListeners()
    {
        return [
            // Public Channel
            "echo:sms-message-status,StatusSMS" => 'smsMessageStatus',
        ];
    }
  
    public function smsMessageStatus(StatusSMS $event)
    {
        dd($event);
        Flux::toast(text: 'SMS Message Failed.', heading: 'SMS Message Status', duration: 0, variant: 'danger', position: "top right");
    }

```
Produces this result

![image](https://github.com/user-attachments/assets/66acf6ba-e632-4998-89ad-3a3ca31217a3)

**Comment by joshhanley** [comment]
@robs007 I think what would be happening because Livewire doesn't know how to restored a serialised version of a Model (which is what Laravel/Reverb will be sending).

I think I would just send the model ID instead and get it from the DB yourself.

**Comment by lrljoe** [comment]
100% agree with Josh, either
a) Send just the Model ID
or
b) Extract the relevant fields

It depends on where events are coming from etc, and what you want to happen for the end user.

Handling it client side can be less intensive on your app if you've got events flowing regularly.

Can you expand on your use case a bit? 
Where is the event being triggered from, 
What's the quantity of events?

For what you shared, listening in a Livewire Component, just to then display the content client side, I'd just handle it all client side, and not worry about that particular Livewire component at all!

**Comment by robs007** [comment]
Thanks,
The event is triggered by a call back from Twilio on an SMS status. I trigger the event if the message fails to send. It should be low use.
I can just send the ID, or really, I just need the name, so maybe I will just send that. Maybe I misinterpreted the documentation's ability to pass the model like pusher does. I would have assumed the pusher was restoring the serialized data.
Thanks

**Comment by robs007** [comment]
Why doesn't this work? I get "Unresolvable dependency resolving [Parameter #0 [ <required> string $name ]] in class App\Events\StatusSMS"

```
StatusSMS::dispatch("john");
```

```
class StatusSMS implements ShouldBroadcastNow
{
    use Dispatchable, InteractsWithSockets, SerializesModels;

    public string $name;

    public function __construct(string $name)

    {
        $this->name = $name;

    }
```

**Reply by lrljoe** [reply]
You don't need to do the property and matching btw 

```
public function __construct(
        public string $name,
    ) {}
```

Worth clearing caches, and restarting your workers/artisan/php process if you haven't already as you may have some cached code.

If you define a string variable in your parent component and dispatch that, does it work?

What else is in that Event from a code perspective?

**Comment by robs007** [comment]
I am still getting the error. I'm dispatching it from Tinkerwell. Could that be the issue? I have tried it several ways.


~~~
use Illuminate\Support\Facades\Notification;
use App\Events\StatusSMS;

$name = "john";

StatusSMS::dispatch($name);
~~~

~~~
$name = "john";
$d = new StatusSMS($name);
$d->dispatch($name);
~~~
~~~
namespace App\Events;

use App\Models\ResStudent;
use App\Models\User;
use Illuminate\Broadcasting\Channel;
use Illuminate\Broadcasting\InteractsWithSockets;
use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
use Illuminate\Contracts\Broadcasting\ShouldBroadcastNow;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Queue\SerializesModels;

class StatusSMS implements ShouldBroadcastNow
{
    use Dispatchable, InteractsWithSockets, SerializesModels;
    
    public function __construct(public string $name)
    {
    }

    /**
     * Get the channels the event should broadcast on.
     *
     * @return array<int, \Illuminate\Broadcasting\Channel>
     */
    public function broadcastOn(): Channel
    {
        return
            new Channel('sms-message-status');
    }

}
~~~

**Comment by robs007** [comment]
I created  a new event and have the same issue. It looks like the data is getting to pusher ok, just not to livewire. In the console I can see this.
Maybe, it's an echo issue.
~~~
pusher-js.js?v=c1f4473d:957 Pusher :  : ["Event recd",{"event":"App\\Events\\SmsMessageStatus","channel":"sms-message-status-channel","data":{"id":"john"}}]
~~~

**Comment by robs007** [comment]
Yeah, it seems to be an Echo loading issue. It may have something to do with herd and local testing. I will move on and look at it again when I'm on a live server. I see it loaded in the JS, but it's not being found.

~~~

var Ko = Gr.exports;
const Go = Vo(Ko);
window.Pusher = Go;
window.Echo = new Xo({
    broadcaster: "reverb",
    key: "laravel-herd",
    wsHost: "reverb-8081.herd.test",
    wsPort: "443",
    wssPort: "443",
    forceTLS: !0,
    enabledTransports: ["ws", "wss"]
});
~~~

~~~
reservations:12680 Uncaught ReferenceError: Echo is not defined
    at reservations:12680:9

~~~
